syncer/syncer.go

```diff
@@ -681,13 +681,24 @@ func (s *Syncer) Run(ctx context.Context) error {
 
 // Close closes the Syncer's net.Listener.
 func (s *Syncer) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.l == nil {
+		return nil // already closed
+	}
 	err := s.l.Close()
+	s.mu.Unlock()
 	s.wg.Wait()
+	s.mu.Lock()
+	s.l = nil
```

Comment:
`Addr()` exposes the listener's address so we have to add an extra `nil` check there as well